File: app/src/main/java/com/botijasoftware/utils/ArrayBuffer.java
package com.botijasoftware.utils;

import java.nio.Buffer;

public abstract class ArrayBuffer {
	protected VertexBufferDefinition vbdefinition;
	//protected Buffer mBuffer;
	protected int maxelements;

	public ArrayBuffer(VertexBufferDefinition vbdefinition, int maxelements) {
		this.vbdefinition = vbdefinition;
		this.maxelements = maxelements;
	}
	
	public abstract Buffer getBuffer();
	
	public void put(int index, float value) {}
	public void put(float[] data, int offset) {}
	public void put(float[] data) {}
	public void put(int index, int value) {}
	public void put(int[] data, int offset) {}
	public void put(int[] data) {}
	public void put(int index, short value) {}
	public void put(short[] data, int offset) {}
	public void put(short[] data) {}
	public void put(int index, byte value) {}
	public void put(byte[] data, int offset) {}
	public void put(byte[] data) {}
	
	public void put(byte value) {}
	public void put(short value) {}
	public void put(int value) {}
	public void put(float value) {}
	
	public void position(int i) {}
	public int position() {return 0;}

}

File: app/src/main/java/com/botijasoftware/utils/ArrayBuffer.kt
package com.botijasoftware.utils

import java.nio.Buffer

abstract class ArrayBuffer(protected var vbdefinition: VertexBufferDefinition, //protected Buffer mBuffer;
                           protected var maxelements: Int) {

    abstract val buffer: Buffer

    open fun put(index: Int, value: Float) {}
    open fun put(data: FloatArray, offset: Int) {}
    open fun put(data: FloatArray) {}
    open fun put(index: Int, value: Int) {}
    open fun put(data: IntArray, offset: Int) {}
    open fun put(data: IntArray) {}
    open fun put(index: Int, value: Short) {}
    open fun put(data: ShortArray, offset: Int) {}
    open fun put(data: ShortArray) {}
    open fun put(index: Int, value: Byte) {}
    open fun put(data: ByteArray, offset: Int) {}
    open fun put(data: ByteArray) {}

    open fun put(value: Byte) {}
    open fun put(value: Short) {}
    open fun put(value: Int) {}
    open fun put(value: Float) {}

    open fun position(i: Int) {}
    open fun position(): Int {
        return 0
    }

}